### PR TITLE
register check activity in temporal for the sync job type

### PR DESCRIPTION
**Pull request by sherifnada**
TL;DR: Temporal has a concept of workers. Workers have types. Each worker type expects only a certain set of activities. It uses this information to serialize/deserialize configurations between the orchestrator and the actual worker runtime. The main issue with the upstream PR was that it didn't register the `CheckConnectionActivityImpl` as a valid activity that could be sent to the `SYNC` job/worker type. This caused the workflow to get confused about how to deserialize the input JSON sent to it. This PR registers the activity type. This problem ceased happening for me during unit test runs. I did not verify this in a proper docker image.

### CODE REVIEW

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/check/connection/CheckConnectionActivityImpl.java [hunk 1]
```diff
@@ -55,9 +55,9 @@ public CheckConnectionActivityImpl(final WorkerConfigs workerConfigs,
     this.airbyteVersion = airbyteVersion;
   }
 
-  public StandardCheckConnectionOutput run(final JobRunConfig jobRunConfig,
-                                           final IntegrationLauncherConfig launcherConfig,
-                                           final StandardCheckConnectionInput connectionConfiguration) {
+  public StandardCheckConnectionOutput check(final JobRunConfig jobRunConfig,
```
Comment (sherifnada) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/check/connection/CheckConnectionActivityImpl.java:58]: this rename was necessary, otherwise we were trying to register two activities whose entrypoint was called `run` which confused temporal

File: airbyte-workers/src/test/java/io/airbyte/workers/temporal/SyncWorkflowTest.java [hunk 2]
```diff
@@ -114,7 +118,7 @@ public void setUp() {
 
   // bundle up all the temporal worker setup / execution into one method.
   private StandardSyncOutput execute() {
-    syncWorker.registerActivitiesImplementations(replicationActivity, normalizationActivity, dbtTransformationActivity, persistStateActivity);
+    syncWorker.registerActivitiesImplementations(checkActivity, replicationActivity, normalizationActivity, dbtTransformationActivity, persistStateActivity);
```
Comment (sherifnada) [inline airbyte-workers/src/test/java/io/airbyte/workers/temporal/SyncWorkflowTest.java:121]: here we register the activity type in the unit test

File: airbyte-workers/src/main/java/io/airbyte/workers/WorkerApp.java [hunk 3]
```diff
@@ -195,11 +195,15 @@ private void registerSync(final WorkerFactory factory) {
...
+    final CheckConnectionActivityImpl checkConnectionActivity =
+        new CheckConnectionActivityImpl(checkWorkerConfigs, checkProcessFactory, secretsHydrator, workspaceRoot, workerEnvironment, logConfigs,
+            jobPersistence, airbyteVersion);
+
     final PersistStateActivityImpl persistStateActivity = new PersistStateActivityImpl(workspaceRoot, configRepository);
 
     final Worker syncWorker = factory.newWorker(TemporalJobType.SYNC.name(), getWorkerOptions(maxWorkers.getMaxSyncWorkers()));
     syncWorker.registerWorkflowImplementationTypes(SyncWorkflowImpl.class);
-    syncWorker.registerActivitiesImplementations(replicationActivity, normalizationActivity, dbtTransformationActivity, persistStateActivity);
+    syncWorker.registerActivitiesImplementations(checkConnectionActivity, replicationActivity, normalizationActivity, dbtTransformationActivity, persistStateActivity);
```
Comment (sherifnada) [inline airbyte-workers/src/main/java/io/airbyte/workers/WorkerApp.java:206]: here we regsiter the activity in the worker app, which is what runs when you build the docker image proper

**Review by evantahler — APPROVED** [review]
❤️
